intellij/src/main/kotlin/motif/intellij/ui/MotifPanel.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2018-2019 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package motif.intellij.ui
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.ui.components.JBTabbedPane
+import motif.core.ResolvedGraph
+import motif.intellij.MotifProjectComponent
+import motif.intellij.hierarchy.ScopePropertyHierarchyBrowser
+import motif.intellij.hierarchy.ScopePropertyHierarchyBrowser.PropertyHierarchyType
+import motif.intellij.hierarchy.ScopeHierarchyBrowser
+import motif.intellij.ScopeHierarchyUtils
+import motif.models.Scope
+import java.awt.BorderLayout
+import javax.swing.JPanel
+import javax.swing.JSplitPane
+
+class MotifPanel(project: Project, graph: ResolvedGraph) : JPanel(), ScopeHierarchyBrowser.Listener, MotifProjectComponent.Listener {
+
+    companion object {
+        const val TAB_NAME_CONSUMES: String = "Consumes"
+        const val TAB_NAME_PROVIDES: String = "Provides"
+        const val TAB_NAME_DEPENDENCIES: String = "Dependencies"
+    }
+
+    private val splitPane: JSplitPane
+    private val tabs: JBTabbedPane
+    private var scopeBrowser: ScopeHierarchyBrowser
```

Comment:
These should all be `val`s since the values aren't expected to change

Reply:
done